packages/databricks-vscode/src/bundle/run/JobRunStatus.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    JobsError,
+    JobsRetriableError,
+} from "@databricks/databricks-sdk/dist/apis/jobs";
+import {Event, EventEmitter} from "vscode";
+import {AuthProvider} from "../../configuration/auth/AuthProvider";
+import {onError} from "../../utils/onErrorDecorator";
+import {BundleRunStatus} from "./BundleRunStatus";
+
+export class JobRunStatus extends BundleRunStatus {
+    readonly type = "jobs";
+
+    private readonly onDidCompleteEmitter = new EventEmitter<void>();
+    onDidComplete: Event<void> = this.onDidCompleteEmitter.event;
+
+    constructor(private readonly authProvider: AuthProvider) {
+        super();
+    }
+
+    parseId(output: string) {
+        if (this.runId !== undefined || this.runState !== "unknown") {
+            return;
+        }
+        const match = output.match(/.*\/run\/(\d*).*/);
+        if (match === null) {
+            return;
+        }
+        this.runId = match[1];
+
+        this.startPolling();
+    }
+
+    @onError({popup: {prefix: "Error starting polling."}})
+    private async startPolling() {
+        if (this.runState !== "unknown") {
+            return;
+        }
+
+        if (this.runId === undefined) {
+            throw new Error("No run id found");
+        }
+
+        const client = this.authProvider.getWorkspaceClient();
+        try {
+            this.runState = "running";
+            await (
+                await client.jobs.getRun({run_id: parseInt(this.runId)})
+            ).wait({
+                onProgress: async (progress) => {
+                    this.data = progress;
+                    this.onDidChangeEmitter.fire();
+                },
+            });
+        } catch (e) {
+            if (e instanceof JobsError || e instanceof JobsRetriableError) {
+                // On progress is already fired for these status updates. We let the listeners handle it.
+                return;
```

Comment:
Are we fine with being stuck in the `running` state here?

Reply:
Default is 10 mins

Reply:
Actually, makes sense. We should change the running state here. The running state here is mostly for polling being running or not, which need not coincide with the actual run happening or not. But no harm in keeping this state consistent.